Cannot `UNNEST` null array fields

```sql
WITH file AS (
  SELECT 1 AS file_id, ARRAY<STRING>["r", "w"] AS modes
  UNION ALL SELECT 2, ARRAY<STRING>["w"]
)
SELECT id,
(SELECT mode FROM UNNEST(modes) AS mode WHERE mode = 'w') IS NOT NULL AS write_mode,
(SELECT mode FROM UNNEST(modes) AS mode WHERE mode = 'r') IS NOT NULL AS read_mode
FROM UNNEST([1, 2, 3]) id
LEFT JOIN file on file.file_id = id
```

### Expected
| id | write_mode | read_mode |
| -- | -- | -- |
1 | true | true
2 | true | false
3 | false | false

### Actual 
| id | write_mode | read_mode |
| -- | -- | -- |
1 | true | true
2 | true | false

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x70 pc=0x100ee1d94]
```

--- a/internal/function_register.go
+++ b/internal/function_register.go
@@ -427,6 +427,9 @@ func RegisterFunctions(conn *sqlite3.SQLiteConn) error {
 		if err != nil {
 			return "", err
 		}
+		if decoded == nil {
+			return "[]", nil
+		}
 		array, err := decoded.ToArray()
 		if err != nil {
 			return "", err

--- a/query_test.go
+++ b/query_test.go
@@ -2515,6 +2515,22 @@ FROM
 				},
 			},
 		},
+		// Regression test for goccy/go-zetasqlite#179
+		{
+			name: "null array scan",
+			query: `
+WITH file AS (
+  SELECT 1 AS file_id, ARRAY<STRING>["r", "w"] AS modes
+  UNION ALL SELECT 2, ARRAY<STRING>["w"]
+)
+SELECT id,
+(SELECT mode FROM UNNEST(modes) AS mode WHERE mode = 'w') IS NOT NULL AS write_mode,
+(SELECT mode FROM UNNEST(modes) AS mode WHERE mode = 'r') IS NOT NULL AS read_mode
+FROM UNNEST([1, 2, 3]) id
+LEFT JOIN file on file.file_id = id`,
+			expectedRows: [][]interface{}{{int64(1), true, true}, {int64(2), true, false}, {int64(3), false, false}},
+		},
+
 		{
 			name: "array_reverse function",
 			query: `

